infrastructures/infrastructure-common/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/ConnectorIaasController.java

```diff
@@ -253,18 +255,20 @@ public String createAzureInfrastructure(String infrastructureId, String clientId
     }
 
     public Set<String> createAwsEc2InstancesWithOptions(String infrastructureId, String instanceTag, String image,
-            int numberOfInstances, int cores, int ram, String spotPrice, String securityGroupNames, String subnetId,
-            String macAddresses, String username, String publicKeyName) {
+            int numberOfInstances, int cores, int ram, String vmType, String spotPrice, String securityGroupNames,
+            String subnetId, String macAddresses, int[] portsToOpen, String username, String publicKeyName) {
```

Comment:
I thought portsToOpen  has type Set<Integer> isn't it ?

Reply:
It was the type Set in AWSEC2CustomizableParameter, because we may need to add the port 22 (to run initial script). When passing to the connector-iaas, since it's immutable, we use its primitive type to facilitate later operations.